src/ast/mod.rs

```diff
@@ -624,6 +590,12 @@ pub enum Expr {
     Identifier(Ident),
     /// Multi-part identifier, e.g. `table_alias.column` or `schema.table.col`
     CompoundIdentifier(Vec<Ident>),
+    /// Multi-part Expression accessing. It's used to represent an access chain from a root expression.
+    /// e.g. `expr[0]`, `expr[0][0]`, or `expr.field1.filed2[1].field3`, ...
+    CompoundExpr {
```

Comment:
Also, it seems to me that the name `Compound` is somewhat misleading in this context

It isn't a compound expression per se, it seems more like an a compoound access / multi level access

What do you think about names like `CompoundAccess` ?(that would also fit nicely with @iffyio 's suggestion at https://github.com/apache/datafusion-sqlparser-rs/pull/1551/files#r1871788945 for `AccessExpr`. Maybe something like 🤔  


```rust
...
    CompoundAccess {
      root: Box<Expr>,
      accesses: Vec<AccessExprs>
...
```

I am not sure I really like `accesses` (too many `s` 🤔 )

Maybe `CompoundFieldAccess` 🤔

Reply:
> I am not sure I really like `accesses` (too many `s` 🤔 )
>
If not prefer `accesses`, how about `access_chain` or `access_path`?

> Maybe `CompoundFieldAccess` 🤔
>
`CompoundFieldAccess` is good to me.